core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -63,80 +66,136 @@ public class SharePartitionManager {
     private final ReplicaManager replicaManager;
     private final Time time;
     private final ShareSessionCache cache;
+    private ConcurrentLinkedQueue<ShareFetchPartitionData> fetchQueue;
+    private final Lock lock;
 
     public SharePartitionManager(ReplicaManager replicaManager, Time time, ShareSessionCache cache) {
         this.replicaManager = replicaManager;
         this.time = time;
         this.cache = cache;
         partitionCacheMap = new ConcurrentHashMap<>();
+        fetchQueue = new ConcurrentLinkedQueue<>();
+        this.lock = new ReentrantLock();
     }
 
     // TODO: Move some part in share session context and change method signature to accept share
     //  partition data along TopicIdPartition.
     public CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionData>> fetchMessages(
-        String groupId,
-        String memberId,
-        FetchParams fetchParams,
-        List<TopicIdPartition> topicIdPartitions) {
+            String groupId,
+            String memberId,
+            FetchParams fetchParams,
+            List<TopicIdPartition> topicIdPartitions) {
```

Comment:
nit: spacing seems correct earlier, maybe you want `) {` in the new line where params end.